File: claude-projects/financeTracker/financetracker/backend/src/infrastructure/api/schemas/projection.py
"""
Pydantic Schemas for Projection Request/Response
"""
from __future__ import annotations

from pydantic import BaseModel, Field
from typing import Optional


class ProjectionPointResponse(BaseModel):
    """Single point in projection data."""

    date: str
    balance: str
    net_change: str

    class Config:
        from_attributes = True


class ProjectionResponse(BaseModel):
    """Complete projection response."""

    scenario: str
    starting_balance: str
    ending_balance: str
    min_balance: str
    max_balance: str
    average_balance: str
    total_change: str
    is_critical: bool
    is_warning: bool
    is_healthy: bool
    is_improving: bool
    num_days: int
    num_negative_days: int
    percentage_negative_days: float
    projection_points: list[ProjectionPointResponse]

    class Config:
        from_attributes = True


class ProjectionQueryRequest(BaseModel):
    """Request parameters for projection query."""

    months: int = Field(6, ge=1, le=12, description="Number of months to project (1-12)")
    scenario: str = Field(
        "realistic",
        description="Scenario type: pessimistic, realistic, or optimistic"
    )

File: claude-projects/financeTracker/financetracker/frontend/src/api/projection.ts
import { apiClient } from './client';
import { ProjectionResponse } from './types';

export const projectionApi = {
  /**
   * Get balance projection for all scenarios
   */
  get: async (months: number = 6, scenario?: 'pessimistic' | 'realistic' | 'optimistic'): Promise<ProjectionResponse> => {
    const params: Record<string, unknown> = {
      months,
    };
    if (scenario) {
      params.scenario = scenario;
    }

    const response = await apiClient.get<ProjectionResponse>('/projection', {
      params,
    });
    return response.data;
  },

  /**
   * Get projection for a specific scenario
   */
  getScenario: async (
    scenario: 'pessimistic' | 'realistic' | 'optimistic',
    months: number = 6
  ): Promise<ProjectionResponse> => {
    return projectionApi.get(months, scenario);
  },
};
